source/mir/ndslice/algorithm.d

```diff
@@ -1884,3 +1885,714 @@ size_t countImpl(alias fun, Slices...)(Slices slices)
     while(!slices[0].empty);
     return ret;
 }
+
+version(unittest)
+{
+    T identity(T)(T x)
+    {
+        return x;
+    }
+}
+
+/++
+The call `eachLower!(fun)(matrix)` applies `fun` to the lower triangle of
+a two-dimensional slice.
+
+The value `k` determines which diagonals will have the function applied:
+For k = 0, the function is also applied to the main diagonal
+For k = 1 (default), only the non-main diagonals below the main diagonal
+will have the function applied.
+For k > 1, fewer diagonals below the main diagonal will have the function
+applied.
+For k < 0, more diagonals above the main diagonal will have the function
+applied.
+
+Params:
+    fun = A function
+    k = adjustment to diagonals (default = 1)
++/
+template eachLower(alias fun, ptrdiff_t k = 1)
```

Comment:
why no to make `k` a runtime parameter?

Reply:
I will make it a run-time parameter. The reason I hadn't is because I couldn't think of a time when I wanted to use `triu` or `tril` without knowing at compile-time how I was going to use it. Since this is more general than those functions, it probably makes sense. 

As an aside, it would be much easier if in D you could have some parameters be available in run-time or compile-time.